src/glass/src/app/pages/announce-page/announce-page.component.ts

```diff
@@ -0,0 +1,145 @@
+
+@Component({
+  selector: 'glass-announce-page',
+  templateUrl: './announce-page.component.html',
+  styleUrls: ['./announce-page.component.scss']
+})
+export class AnnouncePageComponent implements OnInit {
+  @BlockUI()
+  blockUI!: NgBlockUI;
+
+  joining = false;
+  public formGroup: FormGroup;
+
+  constructor(
+    private formBuilder: FormBuilder,
+    private nodesService: NodesService,
+    private notificationService: NotificationService,
+    private pollService: PollService,
+    private router: Router,
+    private statusService: StatusService
+  ) {
+    this.formGroup = this.formBuilder.group({
+      address: [null, [Validators.required, this.addressValidator()]],
+      token: [null, [Validators.required, Validators.pattern(TOKEN_REGEXP)]]
+    });
+  }
+
+  ngOnInit(): void {
+    this.blockUI.resetGlobal();
+    this.statusService.status().subscribe((res: Status) => {
+      if (res.node_stage === StatusStageEnum.joining) {
+        this.joining = true;
+        this.blockUI.start(
+          translate(TEXT('Please wait, joining existing cluster in progress ...'))
+        );
+        this.pollJoiningStatus();
+      }
+    });
+  }
+
+  doJoin(): void {
+    if (this.formGroup.pristine || this.formGroup.invalid) {
+      return;
+    }
+    const values = this.formGroup.value;
+    const handleError = () => {
+      this.joining = false;
+      this.blockUI.stop();
+      this.notificationService.show(TEXT('Failed to join existing cluster.'), {
+        type: 'error'
+      });
+    };
+    this.joining = true;
+    this.blockUI.start(translate(TEXT('Please wait, start joining existing cluster ...')));
+    this.nodesService.join(values).subscribe({
+      next: (success: boolean) => {
+        if (success) {
+          this.blockUI.update(
+            translate(TEXT('Please wait, joining existing cluster in progress ...'))
+          );
+          this.pollJoiningStatus();
+        } else {
+          handleError();
+        }
+      },
+      error: () => handleError()
+    });
+  }
+
+  protected pollJoiningStatus(): void {
+    const handleError = () => {
+      this.joining = false;
+      this.blockUI.stop();
+      this.notificationService.show(TEXT('Failed to join existing cluster.'), {
+        type: 'error'
+      });
+    };
+    this.statusService
+      .status()
+      .pipe(
+        this.pollService.poll(
+          (res: Status) => res.node_stage === StatusStageEnum.joining,
+          undefined,
+          TEXT('Failed to join existing cluster.')
+        )
+      )
+      .subscribe(
+        (res: Status) => {
+          switch (res.node_stage) {
+            case StatusStageEnum.none:
+            case StatusStageEnum.unknown:
+              handleError();
+              break;
+            case StatusStageEnum.ready:
+              this.joining = false;
+              this.blockUI.stop();
+              this.router.navigate(['/installer/join/deployment']);
```

Comment:
I think for now we want to drop the user directly onto the dashboard. There won't be anything to be done during deployment (i.e., disks are automatically assimilated, services can be created through the dashboard, etc)